src/main/java/interface_adapter/signup/club_signup/ClubSignupPresenter.java

```diff
@@ -44,6 +44,15 @@ public void prepareFailView(String error) {
 
     @Override
     public void switchToLoginView() {
+        // Clears the LoginState
+        final LoginState state = loginViewModel.getState();
+        state.setLoginError(null);
+        state.setPassword("");
+        state.setIdentifier("");
+        loginViewModel.setState(state);
+        loginViewModel.firePropertyChanged();
+
```

Comment:
same comment as before. It would be easier, and in my opinion better, to just create a new instance of LoginState and set the LoginViewModel state to this new state.

Reply:
That is valid since it is intuitive. I followed the dependency injection pattern as to not have any hard dependencies for easier testing